torch/csrc/autograd/record_function.cpp

```diff
@@ -26,39 +27,86 @@ void popCallback() {
   }
   start_callbacks.pop_back();
   end_callbacks.pop_back();
+  needs_inputs_flags.pop_back();
   has_callbacks = !start_callbacks.empty();
+  cb_needs_inputs = std::any_of(
+    needs_inputs_flags.begin(), needs_inputs_flags.end(),
+    [](bool v) { return v; });
 }
 
-RecordFunction::RecordFunction(Function* fn, GetPackedInputsCallback cb) {
+TORCH_API RecordFunctionState recordFunctionState() {
```

Comment:
can't it be just one variable? your has_callbacks is already a "derived" variable

Reply:
sounds good, that would be simpler